src/utils/check-pr/cleanup_ad.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+
+# IMPORTANT: Same as check-pr.py must be run from a Linux shell
+
+import json
+import subprocess
+import argparse
+from azure.common.client_factory import get_client_from_cli_profile
+
+from azure.core.credentials import AccessToken
+from azure.identity import ManagedIdentityCredential
+from msgraph.core import GraphClient
+
+
+class AZCliCredThatWorks(object):
```

Comment:
seems like you are working some kind od limitation of the existing az cli. Please add a comment to explain why this is necessary